### ISSUE TITLE
Bluetooth DFU and Event services should be instantiated via C/C++ constructor call rather than config.json

### ISSUE THREAD

**Issue by bluetooth-mdw** [lancaster-university/microbit-dal#171]
These two services are enabled/disabled via entries in config.json (and so is Device Information) whereas the others require an object to be constructed in user application code. This is inconsistent but more importantly gives rise to restrictions with PXT. It should be possible to selectively enable these two services using PXT Blocks but I don't think this can be done with them controlled from config.json as opposed to the object instantiation approach.

Can we change this please?

**Comment by finneyj** [comment]
Yes! I'd also extend to include device include DeviceInformationService too. These were mandatory on request of BBC to support microbit.co.uk, but no reason this should influence other build environments. micorbit.co.uk has its own branch, so we can safely make these optional as per other Bluetooth services.
